src/viam/app/data_client.py

```diff
@@ -478,6 +493,111 @@ async def get_database_connection(self, organization_id: str) -> str:
     async def configure_database_user(self, organization_id: str, password: str) -> None:
         raise NotImplementedError()
 
+    class Dataset:
+        """Class representing a dataset.
+
+        Args:
+            id (str): the id of the dataset.
+            name (str): the name of the dataset.
+            organization_id (str): the organization id of the dataset.
+            time_created (datetime): the time the dataset was created.
+        """
+
+        def __init__(self, id: str, name: str, organization_id: str, time_created: datetime) -> None:
+            self.id = id
+            self.name = name
+            self.organization_id = organization_id
+            self.time_created = time_created
```

Comment:
Given the simplicity of converting the proto `Dataset` type to this representation, I suspect it's gonna be easier for us to maintain (and more consistent with the rest of the SDK) to do away with this shadow type entirely and just return ask users to interact with raw proto `Dataset`s.

As an aside (certainly not necessary for this PR!) we should probably get rid of the `BinaryData` shadow class in here since there's no conversion at all, we're okay with users interacting with raw proto in the python SDK, and we run the risk of falling out of date if the proto representation ever changes.

Reply:
Got it, will remove the shadow type. Thanks for the explanation!
I'll make a ticket for removing `BinaryData` since I don't want to remove it without investigation.

Reply:
Sounds good, thanks!